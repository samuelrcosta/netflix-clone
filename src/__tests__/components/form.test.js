import { render, fireEvent } from '@testing-library/react';
import { Form } from '../../components';
import * as ROUTES from '../../constants/routes';
import { MemoryRouter } from 'react-router-dom';

describe('<Form />', () => {
  it('renders <Form /> with populated data', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <MemoryRouter>
        <Form>
          <Form.Title>Sign In now</Form.Title>

          <Form.Base>
            <Form.Input placeholder="Email address" onChange={() => {}} />

            <Form.Input type="password" autoComplete="off" placeholder="Password" onChange={() => {}} />

            <Form.Submit disabled type="submit">
              Sign In
            </Form.Submit>
          </Form.Base>

          <Form.Text>
            New to Netflix? <Form.Link to={ROUTES.SIGN_UP}>Sign up now.</Form.Link>
          </Form.Text>
          <Form.TextSmall>
            This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
          </Form.TextSmall>
        </Form>
      </MemoryRouter>
    );

    expect(
      getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.")
    ).toBeTruthy();
    expect(getByText('Sign up now.')).toBeTruthy();
    expect(getByText('Sign In now')).toBeTruthy();
    expect(getByText('Sign In').disabled).toBeTruthy();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders <Form /> with an error', () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Wrong credentials.</Form.Error>

        <Form.Base>
          <Form.Input placeholder="Email address" onChange={() => {}} />

          <Form.Input type="password" autoComplete="off" placeholder="Password" onChange={() => {}} />

          <Form.Submit type="submit">Sign In</Form.Submit>
        </Form.Base>
      </Form>
    );

    expect(getByText('Wrong credentials.')).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
