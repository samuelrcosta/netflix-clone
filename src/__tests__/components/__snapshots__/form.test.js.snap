// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders <Form /> with an error 1`] = `
<div
  class="form__Container-sc-7952rh-0 cGcgsn"
>
  <div
    class="form__Error-sc-7952rh-1 dePsoc"
  >
    Wrong credentials.
  </div>
  <form
    class="form__Base-sc-7952rh-2 huEoKC"
  >
    <input
      class="form__Input-sc-7952rh-7 krRUWd"
      placeholder="Email address"
    />
    <input
      autocomplete="off"
      class="form__Input-sc-7952rh-7 krRUWd"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 eEdKSg"
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;

exports[`<Form /> renders <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-7952rh-0 cGcgsn"
>
  <h1
    class="form__Title-sc-7952rh-3 cKBDVR"
  >
    Sign In now
  </h1>
  <form
    class="form__Base-sc-7952rh-2 huEoKC"
  >
    <input
      class="form__Input-sc-7952rh-7 krRUWd"
      placeholder="Email address"
    />
    <input
      autocomplete="off"
      class="form__Input-sc-7952rh-7 krRUWd"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 eEdKSg"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7952rh-4 fvOBTc"
  >
    New to Netflix? 
    <a
      class="form__Link-sc-7952rh-6 thCBk"
      href="/signup"
    >
      Sign up now.
    </a>
  </p>
  <p
    class="form__TextSmall-sc-7952rh-5 drVMew"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
